app/uk/gov/hmrc/taxhistory/services/TaxHistoryMongoCacheService.scala

```diff
@@ -28,50 +28,56 @@ import uk.gov.hmrc.taxhistory.model.api.PayAsYouEarn
 import uk.gov.hmrc.taxhistory.utils.Logging
 import uk.gov.hmrc.time.TaxYear
 
-import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
 /**
   * Uses MongoDB to cache instances of `PayAsYouEarn` for a given NINO and year.
   */
-class TaxHistoryCacheService @Inject()(
+class TaxHistoryMongoCacheService @Inject()(
                               val mongoDbConnection: MongoDbConnection,
                               @Named("mongodb.cache.expire.seconds") expireAfterSeconds: Int,
-                              @Named("mongodb.name") mongoSource: String) extends AnyRef with Logging {
+                              @Named("mongodb.name") mongoSource: String)(implicit ec: ExecutionContext) extends PayeCacheService with Logging {
 
   implicit val mongo: () => DB = mongoDbConnection.db
 
   val cacheRepository = new CacheMongoRepository(mongoSource, expireAfterSeconds, Cache.mongoFormats)
 
-  def createOrUpdate(id: String, key: String, toCache: JsValue): Future[Option[JsValue]] = {
-    cacheRepository.createOrUpdate(Id(id), key, toCache).map(x => x.updateType.savedValue.data.map(_ \ key).map(_.get))
+  def insertOrUpdate(key: (Nino, TaxYear), value: PayAsYouEarn): Future[Option[PayAsYouEarn]] = {
+    val (nino, taxYear) = key
+    val mongoId = nino.nino
```

Comment:
should this be nino.value ?

Reply:
The two are the same (https://github.com/hmrc/domain/blob/eecf5a5169a9f2205cc89def487e0b9121fd5af8/src/main/scala/uk/gov/hmrc/domain/Nino.scala#L27)
I'm not sure which is the 'correct' way of referencing it!

Reply:
Okay, guess it comes down to which is clearer? Do you prefer nino.nino or nino.value ?

Reply:
Maybe `nino.value` is better.

Reply:
Okay, go for that then